### ISSUE TITLE
Docs around symlinks for file provisioner are wrong/incomplete

### ISSUE THREAD

**Issue by ashb** [hashicorp/packer#5129]
The docs https://www.packer.io/docs/provisioners/file.html#symbolic-link-uploads are wrong and wont work in a common case: that of a fresh repo checkout. They show using a local shell provisioner to create a tar file, but if that doesn't exist first time then it will never run (because the file provisioner checks for existance before running the local shell provisioner).. This is the error I get :

```
1 error(s) occurred:

* Bad source './toupload': stat ./toupload: no such file or directory
```

The docs need to be updated as the example in the docs will not work on a fresh checkout of a packer template repo.

(Given I have only use packer from OSs that support symlinks to OSs that support symlink I'm not a huge fan of the "just create them in your shell provisioner. But I guess that ship has sailed. So I'm just asking for the docs to not be misleading.)

As mentioned in #3210, #2877 et al.

**Comment by ashb** [comment]
Or merge #3891 :)

**Comment by ashb** [comment]
Debug log coming.

**Comment by ashb** [comment]
Template:

```json
{
  "builders": [
    {
      "name": "airflow",
      "type": "amazon-ebs",
      "ami_description" : "Airflow AMI",
      "ami_name": "packer-test",
      "instance_type": "t2.small",
      "region": "eu-west-1",
      "source_ami_filter": {
        "filters": {
          "virtualization-type": "hvm",
          "name": "*ubuntu-xenial-16.04-amd64-server-*",
          "root-device-type": "ebs"
        },
        "owners": ["099720109477"],
        "most_recent": true
      }
    }
  ],

  "provisioners":
  [

    {
      "type": "shell-local",
      "command": "mkdir -p toupload; tar -vcf toupload/files.tar files/airflow"
    },
    {
      "destination": "/tmp/",
      "source": "./toupload",
      "type": "file"
    }
  ]
}
```

Debug log: https://gist.github.com/ashb/bc11118262a478c8886878ee4c601a83
```

**Comment by rickard-von-essen** [comment]
Kind of duplicates #2479, but also points out the invalid example for handling symlinks.

**Comment by ilons** [comment]
So, this is still and issue, I just hit this and spent time figuring out why 3 years later.

**Comment by SwampDragons** [comment]
This was addressed by #5624 but never closed. Closing now.
